### PR TITLE
Return explanations along with stattype guesses

### PR DESCRIPTION
**Pull request by asilversempirical**

### CODE REVIEW

File: src/mml_utils.py [hunk 1]
```diff
@@ -52,22 +57,24 @@ def _type_given_vals(vals):
...
-    if len(vals) == 1:
-        return StatType.IGNORE
+    if cardinality == 1:
+        return (StatType.IGNORE, 'Column is constant')
     # Even if the whole column is numerical, if there are only a few distinct
     # values they are very likely enums of a sort.
-    elif len(vals) < 20:
-        return StatType.CATEGORICAL
+    elif cardinality < 20:
+        return (StatType.CATEGORICAL, 'Only %s distinct values' % cardinality)
```
Comment (madeleine-empirical) [inline src/mml_utils.py]: %d

File: src/mml_utils.py [hunk 2]
```diff
@@ -52,22 +57,24 @@ def _type_given_vals(vals):
...
+        return (StatType.NUMERICAL, 'Contains exclusively numbers. %s of them'
+                                    % cardinality)
+    elif cardinality > 1000:
         # That's a lot of values for a categorical.
-        # TODO(asilvers): This seems like a reasonable guess, but an
-        # explanation in the resulting schema would make this more usable.
-        return StatType.IGNORE
+        nonnum = cardinality - len(filter(_numbery, vals))
+        return (StatType.IGNORE, '%s distinct values. %s are non-numeric'
+                                 % (cardinality, nonnum))
```
Comment (madeleine-empirical) [inline src/mml_utils.py:76]: %d

File: src/mml_utils.py [hunk 3]
```diff
@@ -52,22 +57,24 @@ def _type_given_vals(vals):
...
     # Even if the whole column is numerical, if there are only a few distinct
     # values they are very likely enums of a sort.
-    elif len(vals) < 20:
-        return StatType.CATEGORICAL
+    elif cardinality < 20:
+        return (StatType.CATEGORICAL, 'Only %s distinct values' % cardinality)
     elif all(_numbery(v) for v in vals):
-        return StatType.NUMERICAL
-    elif len(vals) > 1000:
+        return (StatType.NUMERICAL, 'Contains exclusively numbers. %s of them'
```
Comment (madeleine-empirical) [inline src/mml_utils.py]: Punctuation? `'Contains exclusively numbers (%d of them).'`

**Comment by madeleine-empirical** [comment]
Minor comments, otherwise LGTM

**Comment by asilversempirical** [comment]
Opened https://github.com/probcomp/bdbcontrib/pull/155

**Pull request closed without merging**